cmd/registry/cmd/label_test.go

```diff
@@ -0,0 +1,199 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/rpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func TestLabel(t *testing.T) {
+	var err error
+
+	projectId := "label-test"
+	projectName := "projects/" + projectId
+	apiId := "sample"
+	apiName := projectName + "/apis/" + apiId
+	versionId := "1.0.0"
+	versionName := apiName + "/versions/" + versionId
+	specId := "openapi.json"
+	specName := versionName + "/specs/" + specId
+
+	// Create a registry client.
+	ctx := context.Background()
+	registryClient, err := connection.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("error creating client: %+v", err)
+	}
+	defer registryClient.Close()
+	// Clear the test project.
+	err = registryClient.DeleteProject(ctx, &rpc.DeleteProjectRequest{
+		Name: projectName,
+	})
+	if err != nil && status.Code(err) != codes.NotFound {
+		t.Fatalf("error deleting test project: %+v", err)
+	}
+	// Create the test project.
+	_, err = registryClient.CreateProject(ctx, &rpc.CreateProjectRequest{
+		ProjectId: projectId,
+		Project: &rpc.Project{
+			DisplayName: "Test",
+			Description: "A test catalog",
+		},
+	})
+	if err != nil {
+		t.Fatalf("error creating project %s", err)
+	}
+	// Create a sample api.
+	_, err = registryClient.CreateApi(ctx, &rpc.CreateApiRequest{
+		Parent: projectName,
+		ApiId:  apiId,
+		Api:    &rpc.Api{},
+	})
+	if err != nil {
+		t.Fatalf("error creating api %s", err)
+	}
+	// Create a sample version.
+	_, err = registryClient.CreateApiVersion(ctx, &rpc.CreateApiVersionRequest{
+		Parent:       apiName,
+		ApiVersionId: versionId,
+		ApiVersion:   &rpc.ApiVersion{},
+	})
+	if err != nil {
+		t.Fatalf("error creating version %s", err)
+	}
+	// Create a sample spec.
+	_, err = registryClient.CreateApiSpec(ctx, &rpc.CreateApiSpecRequest{
+		Parent:    versionName,
+		ApiSpecId: specId,
+		ApiSpec: &rpc.ApiSpec{
+			MimeType: "application/x.openapi;version=3.0.0",
+			Contents: []byte(`{"openapi": "3.0.0", "info": {"title": "test", "version": "v1"}, "paths": {}}`),
+		},
+	})
+	if err != nil {
+		t.Fatalf("error creating spec %s", err)
+	}
+
+	// Add some labels to the test api.
+	rootCmd.SetArgs([]string{"label", apiName, "a=1", "b=2"})
+	if err := labelCmd.Execute(); err != nil {
+		t.Fatalf("Execute() with args %v returned error: %s", rootCmd.Args, err)
+	}
+	api, err := registryClient.GetApi(ctx, &rpc.GetApiRequest{
+		Name: apiName,
+	})
+	if !reflect.DeepEqual(api.Labels, map[string]string{"a": "1", "b": "2"}) {
+		t.Errorf("labels incorrectly set %+v", api.Labels)
+	}
+	// Remove one label and overwrite the other.
```

Comment:
These seem like separate test cases. Consider using table-driven tests for cases that have the same parameters and verification (probably one table per resource type). Alternatively but less extensible, these should at least be wrapped in different subtests using `t.Run("the comment description", func(t *testing.T) { ... })`. More info [here](https://blog.golang.org/subtests).

Reply:
I've restructured these to follow the table-driven style